azavea/district-builder-dtl-pa
# Support deploying to multiple AWS accounts

## Overview

This PR parametrizes parts of the deployment pipeline to support deploying this project to multiple AWS accounts. Additionally, I added a convenient internal Route 53 record for the app server, so that you can ssh from the bastion without looking up the EC2 instance's private IP. Finally, I added `var.cloudfront_aliases` so that we can reach CloudFront from both our domain `pa.districtbuilder.azavea.com` and the client's domain `map.drawthelinespa.org`.

### Changes
- Replace the Azavea AWS credentials in our Travis config with creds from Draw The Lines.
- Add `var.is_subdomain` so that we can conditionally add a DNS record at either the root domain, or a subdomain
- Parametrize S3 bucket prefixes and SSH Key paths in Terraform
- Update defaults in `docker-compose.ci.yml` so that CI defaults to deploying to DTL Staging
- Add App server Internal DNS name
- Add docker-compose.ci.override.yml for local deployments

### Checklist


### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

`var.is_subdomain` - In order to support deploying multiple DistrictBuilder instances into the `districtbuilder.azavea.com` public hosted zone in the root account, we just appended `var.state_name` to the public hosted zone name (i.e. `pa.districtbuilder.azavea.com`). Now that our "root" hosted zone is `pa.districtbuilder.azavea.com`, that naming convention breaks down and we get `pa.pa.districtbuilder.azavea.com`. `var.subdomain` allows us to create a record at the root hosted zone to keep naming consistent.


## Testing Instructions

* Travis has already deployed these changes to staging in the DTL AWS account. See https://staging.pa.districtbuilder.azavea.com
* See [Travis build output](https://travis-ci.org/azavea/district-builder-dtl-pa/builds/411377439?utm_source=github_status&utm_medium=notification)

* Run `scripts/infra`, ensure no changes are necessary. ** Note: before deploying, you'll need to download `district-builder-pa.pem` and the Draw the Lines AWS Credentials off of LastPass **
```bash
$ aws --profile district-builder-pa configure
$ export AWS_PROFILE=district-builder-pa
$ export DB_SETTINGS_BUCKET=district-builder-dtl-staging-config-us-east-1
$ docker-compose -f docker-compose.ci.yml -f docker-compose.ci.override.yml run --rm terraform ./scripts/infra plan
```


Needs azavea/district-builder-core-deployment#9
Closes azavea/district-builder-core-deployment#8
